### ISSUE TITLE
Issues when record_name is empty

### ISSUE THREAD

**Issue by bhean** [cullenmcdermott/terraform-provider-porkbun#20]
Hi, 

First, thanks for share this project! EXTREMELY useful for PorkBun users! 👍🏽 

I'm facing some issues trying to manage the DNS records when those records applies to the "root" domain (when the parameter "name" is empty). I think these issues could be related with the fact that looks like the PorkBun API gets the name of the records with the domain appended, but I'm not sure.

The steps to reproduce the issue in my case are:

- Create a DNS record with an empty name (success):
```
Terraform will perform the following actions:

  # porkbun_dns_record.records will be created
  + resource "porkbun_dns_record" "records" {
      + content = "test.com"
      + domain  = "mydomain.com"
      + id      = (known after apply)
      + ttl     = "600"
      + type    = "ALIAS"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────

Saved the plan to: .terraform/terraform.tfplan

To perform exactly these actions, run the following command to apply:
    terraform apply ".terraform/terraform.tfplan"
Terraform plan built successfully!

Applying Terraform plan:
porkbun_dns_record.records: Creating...
porkbun_dns_record.records: Creation complete after 2s [id=281421583]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

Creation was OK, and I can see the DNS record successfully in the PorkBun dashboard... but if I just re-execute the code again (with NO changes at all, just re-run the Terraform code again) it's not idempotent... the execution fails:
```
Terraform will perform the following actions:

  # porkbun_dns_record.records will be updated in-place
  ~ resource "porkbun_dns_record" "records" {
      ~ id      = "281421583" -> (known after apply)
      - name    = "mydomain.com" -> null
        # (4 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────

Saved the plan to: .terraform/terraform.tfplan

To perform exactly these actions, run the following command to apply:
    terraform apply ".terraform/terraform.tfplan"
Terraform plan built successfully!

Applying Terraform plan:
porkbun_dns_record.records: Modifying... [id=281421583]
╷
│ Error: Error updating the record
│ 
│   with porkbun_dns_record.records,
│   on domains.tf line 18, in resource "porkbun_dns_record" "records":
│   18: resource "porkbun_dns_record" "records" {
│ 
│ Error 400: {"status":"ERROR","message":"Edit error: We were unable to edit
│ the DNS record."}
╵
```
Looks like the "name" parameter retrieved from the PorkBun API has the domain attached, and this new execution try to modify it because the configured "name" in my code is still empty... and it fails.

Thanks in advance,

**Comment by cullenmcdermott** [comment]
Hey! Thanks for raising this. That's definitely not something I tested. I'll take a look this weekend.

**Comment by awadjinn** [comment]
Thank you in advance !

**Comment by cullenmcdermott** [comment]
I believe I have a fix for this. I may add a test and do some other cleanup so not going to release this just yet.

**Comment by cullenmcdermott** [comment]
It's released, but I had a problem initially and reran the release so now the registry says it has an incorrect SHA-256. I'll follow up with the registry team.

**Comment by cullenmcdermott** [comment]
I just deleted 0.1.1 and published 0.1.2 which installs fine. Good luck and let me know if you encounter any other issues!

**Comment by bhean** [comment]
Now it works like a charm! 
Thank you so much for this fast and good fix!

Best regards,
